Week1/Day2/ExerciseXP/code_XP.py

```diff
@@ -0,0 +1,114 @@
+#Exercise 1 : Set
+my_fav_numbers = set([2,3,5,7,8,10,12,17,21,25])
+my_fav_numbers.update([27,30])
+#Remove the last number --- We can do it only if we know the value of rhe "last" item, because set does not have indexes
+# For ex:my_fav_numbers.remove(30)
+
+friend_fav_numbers = set([3, 11, 18])
+# Concatenate my_fav_numbers and friend_fav_numbers to a new variable called our_fav_numbers
+our_fav_numbers = my_fav_numbers|friend_fav_numbers
+print(our_fav_numbers)
+
+
+#Exercise 2: Tuple
+#Directly no, but you make use list function to add new int and then formate it back to tuple()
+
+#Exercise 3: List
+basket = ["Banana", "Apples", "Oranges", "Blueberries"]
+basket.remove("Banana")
+basket.remove("Blueberries")
+basket.append("Kiwi")
+basket.insert(0,"Apples")
+print(basket.count("Apples"))
+basket.clear()
+print(basket)
+
+#Exercise 4: Floats
+#A float, short for "floating-point number," is a data type used in python to represent
+# decimal numbers. Unlike integers, which represent whole numbers without a fractional part,
+# floats can represent numbers with decimal points
+
+#creating a list of floats
+num = 1.5
+floats_list=[]
+for i in range(8):
+    floats_list.append(num)
+    num+=0.5
+print(floats_list)
+
+#or using list comprehensions
+floats_list = [i/2 for i in range(3,11)]
+
+#Exercise 5: For Loop
+for i in range(1,21):
+    print(i)
+print()
+for i in range(1,21):
+    if i % 2 == 0:
```

Comment:
what is the difference between a number and its index?

Reply:
So, the difference between the number and its index is the numerical value of the index itself, which represents the position of the number within the sequence. But here we defined the index range that starts from 1 to 20, so in this case theay are equal. But if we made a list before loop for we will have another situation:
list = [i for i in range(1,21)]
for i in range(len(list)):
         if i % 2 == 0:
               print(list[i])